### PR TITLE
Refactor check migration, change to use async job

### PR DESCRIPTION
**Pull request by azexcy**
Changes proposed in this pull request:
  - Refactor check migration, change to use async job
  - Add unit test 
  - Update IT case

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have passed maven check: `mvn clean install -B -T2C -DskipTests -Dmaven.javadoc.skip=true -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-api/src/main/java/org/apache/shardingsphere/data/pipeline/api/config/job/ConsistencyCheckJobConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    private final String jobId;
+    
+    private final String referredJobId;
+    
+    private final String algorithmTypeName;
+    
+    @Override
+    public String getSourceDatabaseType() {
+        return null;
+    }
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-api/src/main/java/org/apache/shardingsphere/data/pipeline/api/config/job/ConsistencyCheckJobConfiguration.java:43]: It's better to throw UnsupportedOperationException

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-api/src/main/java/org/apache/shardingsphere/data/pipeline/api/pojo/CreateConsistencyCheckJobParameter.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * Create consistency check job parameter.
+ */
+@Data
+@RequiredArgsConstructor
+public final class CreateConsistencyCheckJobParameter {
+    
+    private final String jobId;
+    
+    private final String algorithmTypeName;
+}
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-api/src/main/java/org/apache/shardingsphere/data/pipeline/api/pojo/CreateConsistencyCheckJobParameter.java:33]: Seems algorithm properties is missed

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/exception/job/PipelineJobHasAlreadyExistedException.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * Pipeline job has already existed exception.
+ */
+public final class PipelineJobHasAlreadyExistedException extends PipelineSQLException {
+    
+    private static final long serialVersionUID = 2854259384634892428L;
+    
+    public PipelineJobHasAlreadyExistedException(final String jobId) {
+        super(XOpenSQLState.GENERAL_ERROR, 81, "Job `%s` has already existed", jobId);
+    }
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/exception/job/PipelineJobHasAlreadyExistedException.java:32]: Error code `81` is already used by `PipelineJobHasAlreadyStartedException`

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckChangedJobConfigurationProcessor.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            return;
+        }
+        switch (eventType) {
+            case ADDED:
+            case UPDATED:
+                if (PipelineJobCenter.isJobExisting(jobId)) {
+                    log.info("{} added to executing jobs failed since it already exists", jobId);
+                } else {
+                    log.info("{} executing jobs", jobId);
+                    CompletableFuture.runAsync(() -> execute(jobConfigPOJO), PipelineContext.getEventListenerExecutor());
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckChangedJobConfigurationProcessor.java:53]: `whenComplete` should be added for `CompletableFuture.runAsync`

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckChangedJobConfigurationProcessor.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                break;
+        }
+    }
+    
+    private void execute(final JobConfigurationPOJO jobConfigPOJO) {
+        ConsistencyCheckJob job = new ConsistencyCheckJob();
+        PipelineJobCenter.addJob(jobConfigPOJO.getJobName(), job);
+        OneOffJobBootstrap oneOffJobBootstrap = new OneOffJobBootstrap(PipelineAPIFactory.getRegistryCenter(), job, jobConfigPOJO.toJobConfiguration());
+        oneOffJobBootstrap.execute();
+        job.setOneOffJobBootstrap(oneOffJobBootstrap);
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckChangedJobConfigurationProcessor.java:70]: It's better to put `job.setOneOffJobBootstrap(oneOffJobBootstrap);` before `oneOffJobBootstrap.execute();`

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJob.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        jobAPI.persistJobItemProgress(jobItemContext);
+        String referredJobId = consistencyCheckJobConfig.getReferredJobId();
+        log.info("execute consistency check, job id:{}, referred job id:{}", checkJobId, referredJobId);
+        JobType jobType = PipelineJobIdUtils.parseJobType(referredJobId);
+        InventoryIncrementalJobPublicAPI jobPublicAPI = PipelineJobPublicAPIFactory.getInventoryIncrementalJobPublicAPI(jobType.getTypeName());
+        Map<String, DataConsistencyCheckResult> dataConsistencyCheckResult;
+        if (StringUtils.isBlank(consistencyCheckJobConfig.getAlgorithmTypeName())) {
+            dataConsistencyCheckResult = jobPublicAPI.dataConsistencyCheck(referredJobId);
+        } else {
+            dataConsistencyCheckResult = jobPublicAPI.dataConsistencyCheck(referredJobId, consistencyCheckJobConfig.getAlgorithmTypeName(), null);
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJob.java:72]: The 3rd parameter `algorithmProps` is missed

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPI.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+package org.apache.shardingsphere.data.pipeline.scenario.consistencycheck;
+
+import org.apache.shardingsphere.data.pipeline.api.ConsistencyCheckJobPublicAPI;
+import org.apache.shardingsphere.data.pipeline.core.api.InventoryIncrementalJobAPI;
+
+/**
+ * Consistency check job API.
+ */
+public interface ConsistencyCheckJobAPI extends ConsistencyCheckJobPublicAPI, InventoryIncrementalJobAPI {
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPI.java:26]: It should not extends `InventoryIncrementalJobAPI`, `PipelineJobAPI` could be used, maybe plus `PipelineJobItemAPI`

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+/**
+ * Consistency check job API impl.
+ */
+@Slf4j
+public final class ConsistencyCheckJobAPIImpl extends AbstractInventoryIncrementalJobAPIImpl implements ConsistencyCheckJobAPI {
+    
+    @Override
+    public JobType getJobType() {
+        return JobType.CONSISTENCY_CHECK;
+    }
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java:61]: `getJobType()` could be put to the bottom of class

Comment (sandynz) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java:61]: And also `getJobClassName()`

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Consistency check job API impl.
+ */
+@Slf4j
+public final class ConsistencyCheckJobAPIImpl extends AbstractInventoryIncrementalJobAPIImpl implements ConsistencyCheckJobAPI {
+
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java:57]: It should not extends `AbstractInventoryIncrementalJobAPIImpl`, `AbstractPipelineJobAPIImpl` could be used

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobId.java [hunk 10]
```diff
@@ -35,15 +32,31 @@ public final class ConsistencyCheckJobId extends AbstractPipelineJobId {
     
     public static final String CURRENT_VERSION = "01";
     
-    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyyMMddHHmm");
+    private static final int MAX_CONSISTENCY_CHECK_VERSION = 9;
     
     private final String pipelineJobId;
     
-    private final String createTimeMinutes;
+    private final Integer consistencyCheckVersion;
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobId.java:39]: `Integer` could be `int`

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobId.java [hunk 11]
```diff
@@ -35,15 +32,31 @@ public final class ConsistencyCheckJobId extends AbstractPipelineJobId {
...
+    /**
+     * Get consistency check version.
+     *
+     * @param consistencyCheckJobId consistency check job id.
+     * @return consistency check version
+     */
+    public static int getConsistencyCheckVersion(final @NonNull String consistencyCheckJobId) {
+        String versionString = consistencyCheckJobId.substring(consistencyCheckJobId.length() - 1);
+        int version = Integer.parseInt(versionString);
+        return version > MAX_CONSISTENCY_CHECK_VERSION ? 0 : version;
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobId.java:60]: It's better not change version any more after it's constructed

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobId.java (hunk 10 above)
Comment (sandynz) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobId.java:39]: And could we rename it to sequence etc?

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobItemContext.java [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+import java.util.LinkedList;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Consistency check job item context.
+ */
+@Getter
+@Setter
+@Slf4j
+public final class ConsistencyCheckJobItemContext implements InventoryIncrementalJobItemContext {
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobItemContext.java:41]: It should not implements `InventoryIncrementalJobItemContext`, `PipelineJobItemContext` could be used

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/metadata/node/PipelineMetaDataNodeTest.java [hunk 13]
```diff
@@ -72,12 +72,12 @@ public void assertGetJobConfigPath() {
...
-        assertThat(PipelineMetaDataNode.getCheckLatestResultPath(jobId), is(jobCheckRootPath + "/latest_result"));
+        assertThat(PipelineMetaDataNode.getCheckLatestJobIdPath(jobId), is(jobCheckRootPath + "/latest_job_id"));
     }
     
     @Test
-    public void assertGetCheckLatestDetailedResultPath() {
-        assertThat(PipelineMetaDataNode.getCheckLatestDetailedResultPath(jobId), is(jobCheckRootPath + "/latest_detailed_result"));
+    public void assertgetCheckJobResultPath() {
+        assertThat(PipelineMetaDataNode.getCheckJobResultPath(jobId, "j02fx123"), is(jobCheckRootPath + "/job_ids/j02fx123"));
     }
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/metadata/node/PipelineMetaDataNodeTest.java:81]: These 2 methods name could be updated

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/migration/AbstractMigrationITCase.java [hunk 14]
```diff
@@ -159,10 +162,20 @@ protected String getJobIdByTableName(final String tableName) {
...
+    protected void assertCheckMigrationSuccess(final String jobId, final String algorithmType) throws SQLException {
+        proxyExecuteWithLog(String.format("CHECK MIGRATION '%s' BY TYPE (NAME='%s')", jobId, algorithmType), 0);
+        List<Map<String, Object>> checkJobResults = Collections.emptyList();
+        for (int i = 0; i < 10; i++) {
+            checkJobResults = queryForListWithLog(String.format("SHOW MIGRATION CHECK STATUS '%s'", jobId));
+            if (null != checkJobResults && !checkJobResults.isEmpty()) {
+                break;
+            }
+            ThreadUtil.sleep(3, TimeUnit.SECONDS);
+        }
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/migration/AbstractMigrationITCase.java:174]: Is for loop and sleep still needed?

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIFactoryTest.java [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public final class ConsistencyCheckJobAPIFactoryTest {
+    
+    @Test
+    public void assertGetInstance() {
+        assertThat(ConsistencyCheckJobAPIFactory.getInstance(), instanceOf(ConsistencyCheckJobAPIImpl.class));
+    }
+}
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIFactoryTest.java:31]: Other API factory need unit test too for ConsistencyCheckJobAPI

**Review by sandynz — CHANGES_REQUESTED** [review]

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/migration/AbstractMigrationITCase.java (hunk 14 above)
Comment (azexcy) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/migration/AbstractMigrationITCase.java:174]: Beacuse `check migration` DistSQL check data async, need wait it finished.

Comment (sandynz) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/migration/AbstractMigrationITCase.java:174]: OK

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-api/src/main/java/org/apache/shardingsphere/data/pipeline/api/check/consistency/yaml/YamlDataConsistencyCheckResult.java [hunk 16]
```diff
@@ -33,9 +33,9 @@
 @Getter
 @Setter
 @ToString
-public final class YamlDataConsistencyCheckResultConfiguration {
+public final class YamlDataConsistencyCheckResult {
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-api/src/main/java/org/apache/shardingsphere/data/pipeline/api/check/consistency/yaml/YamlDataConsistencyCheckResult.java:36]: `YamlDataConsistencyCheckResult` has different hierarchy with `DataConsistencyCheckResult`, it's better to improve it

**Review by sandynz — CHANGES_REQUESTED** [review]

**Comment by azexcy** [comment]
check result persist value

```
bmsql_config: |
  contentCheckResult:
    matched: true
  countCheckResult:
    matched: true
    sourceRecordsCount: 5
    targetRecordsCount: 5
```

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJob.java [hunk 17]
```diff
@@ -74,7 +74,7 @@ public void execute(final ShardingContext shardingContext) {
         }
         Map<String, String> checkResultMap = new LinkedHashMap<>();
         for (Entry<String, DataConsistencyCheckResult> entry : dataConsistencyCheckResult.entrySet()) {
-            YamlDataConsistencyCheckResultValue checkResult = CHECK_RESULT_SWAPPER.swapToYamlConfiguration(entry.getValue());
+            YamlDataConsistencyCheckResult checkResult = CHECK_RESULT_SWAPPER.swapToYamlConfiguration(entry.getValue());
             checkResultMap.put(entry.getKey(), YamlEngine.marshal(checkResult));
         }
         PipelineAPIFactory.getGovernanceRepositoryAPI().persistCheckJobResult(referredJobId, checkJobId, YamlEngine.marshal(checkResultMap));
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJob.java:80]: Could we put marshal and unmarshal to the same class, e.g. GovernanceRepositoryAPIImpl

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java [hunk 18]
```diff
@@ -110,10 +110,10 @@ public Map<String, DataConsistencyCheckResult> getLatestDataConsistencyCheckResu
     
     @Override
     public void persistJobItemProgress(final PipelineJobItemContext jobItemContext) {
-        DataCheckJobItemProgress progress = new DataCheckJobItemProgress();
+        ConsistencyCheckJobProgress progress = new ConsistencyCheckJobProgress();
         progress.setStatus(jobItemContext.getStatus());
-        YamlDataCheckJobProgress yamlDataCheckJobProgress = PROGRESS_SWAPPER.swapToYamlConfiguration(progress);
-        PipelineAPIFactory.getGovernanceRepositoryAPI().persistJobItemProgress(jobItemContext.getJobId(), jobItemContext.getShardingItem(), YamlEngine.marshal(yamlDataCheckJobProgress));
+        YamlConsistencyCheckJobProgress yamlConsistencyCheckJobProgress = PROGRESS_SWAPPER.swapToYamlConfiguration(progress);
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java:115]: `yamlConsistencyCheckJobProgress` could be `yamlJobProgress`.

`progress` could be `jobProgress`.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java [hunk 19]
```diff
@@ -122,15 +122,15 @@ public PipelineJobItemProgress getJobItemProgress(final String jobId, final int
         if (StringUtils.isBlank(progress)) {
             return null;
         }
-        DataCheckJobItemProgress dataCheckJobItemProgress = PROGRESS_SWAPPER.swapToObject(YamlEngine.unmarshal(progress, YamlDataCheckJobProgress.class, true));
-        DataCheckJobItemProgress result = new DataCheckJobItemProgress();
-        result.setStatus(dataCheckJobItemProgress.getStatus());
+        ConsistencyCheckJobProgress consistencyCheckJobProgress = PROGRESS_SWAPPER.swapToObject(YamlEngine.unmarshal(progress, YamlConsistencyCheckJobProgress.class, true));
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java:125]: `consistencyCheckJobProgress` could be `jobProgress`

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java [hunk 20]
```diff
@@ -122,15 +122,15 @@ public PipelineJobItemProgress getJobItemProgress(final String jobId, final int
...
+        ConsistencyCheckJobProgress result = new ConsistencyCheckJobProgress();
+        result.setStatus(consistencyCheckJobProgress.getStatus());
         return result;
     }
     
     @Override
     public void updateJobItemStatus(final String jobId, final int shardingItem, final JobStatus status) {
-        DataCheckJobItemProgress jobItemProgress = (DataCheckJobItemProgress) getJobItemProgress(jobId, shardingItem);
+        ConsistencyCheckJobProgress jobItemProgress = (ConsistencyCheckJobProgress) getJobItemProgress(jobId, shardingItem);
         if (null == jobItemProgress) {
```
Comment (sandynz) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/consistencycheck/ConsistencyCheckJobAPIImpl.java:134]: `jobItemProgress` could be `jobProgress`

**Review by sandynz — CHANGES_REQUESTED** [review]

**Review by sandynz — APPROVED** [review]

**Pull request merged**